haystack/nodes/retriever/link_content.py

```diff
@@ -0,0 +1,197 @@
+    Extracts content from the response text using the boilerpy3 extractor.
+    """
+    extractor = extractors.ArticleExtractor(raise_on_failure=False)
+    try:
+        extracted_content = extractor.get_content(response.text)
+        return extracted_content
+    except Exception:
+        return None
+
+
+def pdf_content_handler(response: Response) -> Optional[str]:
+    # TODO: implement this
+    return None
+
+
+class LinkContentRetriever(BaseRetriever):
+    """
+    LinkContentRetriever fetches content from a URL and converts it into a list of Document objects.
+
+    LinkContentRetriever supports the following content types:
+    - HTML
+
+    """
+
+    def __init__(self, pre_processor: Optional[PreProcessor] = None):
+        """
+        Creates a LinkContentRetriever instance.
+        :param pre_processor: PreProcessor to apply to the extracted text
+        """
+        super().__init__()
+        self.pre_processor = pre_processor
+        self.content_handlers: Dict[str, Callable] = {"html": html_content_handler, "pdf": pdf_content_handler}
+
+    def __call__(self, url: str, timeout: int = 3, doc_kwargs: Optional[dict] = None) -> List[Document]:
+        """
+        Fetches content from a URL and converts it into a list of Document objects.
+        :param url: URL to fetch content from.
+        :param timeout: Timeout in seconds for the request.
+        :param doc_kwargs: Optional kwargs to pass to the Document constructor.
+        :return: List of Document objects.
+        """
+        if not url or not self._is_valid_url(url):
+            raise InvalidURL("Invalid or missing URL: {}".format(url))
+
+        doc_kwargs = doc_kwargs or {}
+        extracted_doc = {"url": url}
+
+        response = self._get_response(url, timeout)
+        if not response:
+            return []
+
+        # will handle non-HTML content types soon, add content type resolution here
+        handler = "html"
+        if handler in self.content_handlers:
+            extracted_content = self.content_handlers[handler](response)
+            if extracted_content:
+                extracted_doc.update({"text": extracted_content})
+            else:
+                logger.debug("Couldn't extract content from URL %s, using content handler %s.", url, handler)
+                return []
+
+        extracted_doc.update(doc_kwargs)
+        document = Document.from_dict(extracted_doc, field_map={"text": "content"})
+        document.meta["timestamp"] = int(datetime.utcnow().timestamp())
+
+        if self.pre_processor:
+            return self.pre_processor.process(documents=[document])
+
+        return [document]
+
+    def retrieve(
+        self,
+        query: str,
+        filters: Optional[FilterType] = None,
+        top_k: Optional[int] = None,
+        index: Optional[str] = None,
+        headers: Optional[Dict[str, str]] = None,
+        scale_score: Optional[bool] = None,
+        document_store: Optional[BaseDocumentStore] = None,
```

Comment:
The fact that almost none of the common retriever parameters are used, suggests to me that _this may not be a Retriever_.

Our definition of a Retriever is

> (something that) Sweeps through a document store and returns a set of candidate documents that are relevant to the query.

Speaking of a WebRetriever, this definition can still be applied (but our knowledge base is not a document store, but the web).

`LinkContentRetriever` could become something like `LinkDownloader`...
WDYT?

Reply:
Yeah, you are right. We don't have to make it a Retriever.